### PR TITLE
The website is 18comic.vip, Chinese

### PR DESCRIPTION
**Pull request by ZongerZY**
The reason why I submitted the two versions is: if the Internet user in Mainland China needs to access the URL 18comic2.biz to log in to the website, if it is a non-Mainland China Internet user needs to access 18comic.vip, I cannot judge the user's IP,So it’s up to the viewer to choose.

### CODE REVIEW

**Comment by AbdullahM0hamed** [comment]
I gotta get this joke in before you manage to fix it: PRs are an illusion.

**Comment by ZongerZY** [comment]
> I gotta get this joke in before you manage to fix it: PRs are an illusion.

What does "PRs is an illusion" mean? My English is not very good, it is difficult to understand

**Comment by arkon** [comment]
Just ignore his bad jokes.

**Comment by AbdullahM0hamed** [comment]
Dictionary.com defines illusion to be `something that deceives by producing a false or misleading impression of reality` (but I'm guessing you know that bit), and I use it to mean x isn't/aren't real.

PR means pull request, and the joke is in reference to the fact that at the moment this request has 0 additions and 0 deletions. 

And yes, my jokes are always this bad.

File: src/zh/jinmandalu/src/eu/kanade/tachiyomi/extension/zh/jinmandalu/Jinmandalu.kt [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        genre = selectDetailsStatusAndGenre(document, 0)
+        status = selectDetailsStatusAndGenre(document, 1).trim()!!.toInt()
+        description = document.select("div.p-t-5.p-b-5").get(7).text() + " >>>插件开发者[twitter:ZongerZY][GitHub:ZongerZY][Discord:ZongerZY#3770],开发者将负责开发中文扩展插件,希望大家关注并支持开发者<<<"
+    }
+
+    // 查询作者信息
+    private fun selectAuthor(document: Document): String {
+        var element = document.select("div.tag-block").get(9)
+        if (element.select("a").size == 0) {
+            return "(插件):ZongerZY"
```
Comment (CarlosEsco) [inline src/zh/jinmandalu/src/eu/kanade/tachiyomi/extension/zh/jinmandalu/Jinmandalu.kt:87]: Why are you returning your username if author is empty

File: src/zh/jinmanhaiwai/src/eu/kanade/tachiyomi/extension/zh/jinmanhaiwai/Jinmanhaiwai.kt [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        determineChapterInfo(document)
+        title = document.select("div.panel-heading").select("div.pull-left").first().text()
+        thumbnail_url = document.select("img.lazy_img.img-responsive").attr("src").split("\\?")[0]
+        author = selectAuthor(document)
+        artist = "插件开发者:ZongerZY"
+        genre = selectDetailsStatusAndGenre(document, 0)
+        status = selectDetailsStatusAndGenre(document, 1).trim()!!.toInt()
+        description = document.select("div.p-t-5.p-b-5").get(7).text() + " >>>插件开发者[twitter:ZongerZY][GitHub:ZongerZY][Discord:ZongerZY#3770],开发者将负责开发中文扩展插件,希望大家关注并支持开发者<<<"
```
Comment (CarlosEsco) [inline src/zh/jinmanhaiwai/src/eu/kanade/tachiyomi/extension/zh/jinmanhaiwai/Jinmanhaiwai.kt:80]: same why are you adding your github and discord info

**Comment by ZongerZY** [comment]
> Dictionary.com defines illusion to be `something that deceives by producing a false or misleading impression of reality` (but I'm guessing you know that bit), and I use it to mean x isn't/aren't real.
> 
> PR means pull request, and the joke is in reference to the fact that at the moment this request has 0 additions and 0 deletions.
> 
> And yes, my jokes are always this bad.

I just understand what you mean now, because I am using gitHub's pull request for the first time. I uploaded the file twice from my local computer and found that the first production file has been deleted, hahaha

File: src/zh/jinmandalu/src/eu/kanade/tachiyomi/extension/zh/jinmandalu/Jinmandalu.kt (hunk 1 above)
Comment (AbdullahM0hamed) [reply src/zh/jinmandalu/src/eu/kanade/tachiyomi/extension/zh/jinmandalu/Jinmandalu.kt]: Don't you know he wrote all the comics that have no author names, smh

File: src/zh/jinmanhaiwai/src/eu/kanade/tachiyomi/extension/zh/jinmanhaiwai/Jinmanhaiwai.kt (hunk 2 above)
Comment (AbdullahM0hamed) [reply src/zh/jinmanhaiwai/src/eu/kanade/tachiyomi/extension/zh/jinmanhaiwai/Jinmanhaiwai.kt]: He is in the plot of all the comics.

**Comment by ZongerZY** [comment]
> factory

It means that I should delete an extension, jinmandalu or jinmanhaiwai, and try to make all use one factory

**Comment by ZongerZY** [comment]
> This code seems to be same you should do a factory for it instead of 2 duplicates. Look at Madara and Foolslide for examples

It means that I should delete an extension, jinmandalu or jinmanhaiwai, and try to make all use one factory

**Comment by AbdullahM0hamed** [comment]
Not sure if that was a question or a statement - but yes, that's what should be done.

**Comment by ZongerZY** [comment]
> Not sure if that was a question or a statement - but yes, that's what should be done.

I just used the IPs of the United States, Chinese mainland, Taiwan, Hong Kong, Japan, and Singapore to access the website, and found that the Japanese IP cannot access the two domain names 18comic.vip and 18comic2.biz, but other regions can access 18comic2.biz. Access, so I decided to delete the Jinmanhaiwai extension, just keep one

**Comment by ZongerZY** [comment]
> This code seems to be same you should do a factory for it instead of 2 duplicates. Look at Madara and Foolslide for examples

I have completed the modification, kept one extension, and deleted the illegal information

File: src/zh/jinmanhaiwai/src/eu/kanade/tachiyomi/extension/zh/jinmanhaiwai/Jinmanhaiwai.kt (hunk 2 above)
Comment (lmj0011) [reply src/zh/jinmanhaiwai/src/eu/kanade/tachiyomi/extension/zh/jinmanhaiwai/Jinmanhaiwai.kt]: I was curious about this as well. I guess we'll never know, huh @ZongerZY

Comment (AbdullahM0hamed) [reply src/zh/jinmanhaiwai/src/eu/kanade/tachiyomi/extension/zh/jinmanhaiwai/Jinmanhaiwai.kt]: He is the description

Comment (AbdullahM0hamed) [reply src/zh/jinmanhaiwai/src/eu/kanade/tachiyomi/extension/zh/jinmanhaiwai/Jinmanhaiwai.kt]: And the description is him.

Comment (lmj0011) [reply src/zh/jinmanhaiwai/src/eu/kanade/tachiyomi/extension/zh/jinmanhaiwai/Jinmanhaiwai.kt]: lol, we know what he was trying to do.

```
The plug-in developer [twitter:ZongerZY][GitHub:ZongerZY][Discord:ZongerZY#3770], the developer will be responsible for the development of Chinese extension plug-ins, I hope everyone will pay attention to and support the developer<<<"
```

Comment (AbdullahM0hamed) [reply src/zh/jinmanhaiwai/src/eu/kanade/tachiyomi/extension/zh/jinmanhaiwai/Jinmanhaiwai.kt]: *hmmmmmmmmmmmmmmmmmmmmmmmm*

**Comment by calion-m** [comment]
@ZongerZY Bad news to me, because I have also written an extension for this site...

**Pull request merged**